app/common/lib/bookmarkFoldersUtil.js

```diff
@@ -117,11 +118,32 @@ const isMoveAllowed = (folderList, sourceDetail, destinationDetail) => {
   return true
 }
 
+const buildFolder = (folderDetails, folders) => {
+  folderDetails = makeImmutable(folderDetails)
+  let key = folderDetails.get('folderId')
+
+  if (!folderDetails.has('folderId')) {
+    key = getNextFolderId(folders)
+  }
+
+  return makeImmutable({
+    title: folderDetails.get('title'),
+    folderId: ~~key,
```

Comment:
Correct this is used to convert from possible string to integer. This conversion is a lot faster then `parseInt` or `parseFloat`. Why I changed this conversion in #10610 is because we need negative number (other bookmarks is -1) and this conversion only converts in positive integer.